### ISSUE TITLE
Feature request: string.join(".")

### ISSUE THREAD

**Issue by vitalfadeev** [godotengine/godot#7657]
**Feature request:**
Want method `StringArray.join(".")`
Like a PHP implode: http://php.net/manual/en/function.implode.php

**example:**
Usage:
```
    var splitted = ["mpv", "open", "dvd"]
    var joined = splitted.join(".")
    print(joined)
```

Output:
```
    "mpv.open.dvd"

```

**Comment by AlexHolly** [comment]
Something like this?

```
func _ready():
	print(join(PoolStringArray(["mpv", "open", "dvd"]),"."))

func join(array,filler):
	var rs = ""
		
	for s in array:
		rs += s+filler

	rs.erase( rs.length()-1, 1)
	return rs
```

I think it won't work only with ["mpv", "open", "dvd"] because this is an Array.

Adding join to PoolStringArray will result in this example.

```
var splitted = ["mpv", "open", "dvd"]
var joined = PoolStringArray(splitted).join(".")
print(joined)
```

So what do you think Array or PoolStringArray  or ...?

**Comment by vitalfadeev** [comment]
```
var splitted = ["mpv", "open", "dvd"]
var joined = PoolStringArray(splitted).join(".")
print(joined)
```

Look good.
I like it.

Why not `StringArray.join(".")` ?
See:
```
print ("a.b.c".split("."))
["a", "b", "c"]

print (typeof("a.b.c".split(".")))
25 
```
25 - TYPE_STRING_ARRAY 
...and StringArray logically neater.

What is PoolStringArray? 
Not found it in documentation of v2.1.2. New feature?

**Comment by AlexHolly** [comment]
I am using 3.0 Master right now.
String.split returns an PoolStringArray in 3.0

I think PoolStringArray  is now 25 - TYPE_STRING_ARRAY

The thing is when you create the Array like this it will always be a simple Array

```
var splitted = ["mpv", "open", "dvd"] -> TYPE_ARRAY = 21 
```

**Comment by vitalfadeev** [comment]
Hm... I love clean syntax.
Like a `["mpv", "open", "dvd"].split(".")`
It will be good.

It conflict with something?

**Comment by vitalfadeev** [comment]
I want 3.0 as fast as possible. Yes. )

**Comment by bojidar-bg** [comment]
@vitalfadeev This issue is not solved yet, but just finished with discussion. We still need someone to implement it :wink:

To clarify:
We want `PoolStringArray` to get a `join` method, which accepts a string parameter for the delimiter, and returns the joined string:
```c++
String PoolStringArray::join(String delimiter) {...}
```

**Comment by akien-mga** [comment]
Fixed by #7882.
